tools/cat2lisp.ml

```
@@ -0,0 +1,283 @@
+    ]
+
+and of_binding (loc, pat, exp) =
+  catsym_alist [
+      ("LOC", of_loc loc);
+      ("PAT", of_pat pat);
+      ("EXP", of_exp exp);
+    ]
+
+
+
+let of_do_test x =
+  key (match x with Acyclic -> "ACYCLIC" | Irreflexive -> "IRREFLEXIVE" | TestEmpty -> "TESTEMPTY" )
+
+let of_test x =
+  of_list (match x with
+           | Yes test -> [ key "T_YES"; of_do_test test ]
+           | No test  -> [ key "T_NO"; of_do_test test ])
+
+let of_test_type x =
+  key (match x with Flagged -> "FLAGGED" | UndefinedUnless -> "UNDEFINEDUNLESS" | Check -> "CHECK" | Assert -> "ASSERT" )
+
+let of_app_test (loc, pos, test, exp, str) =
+    catsym_alist [
+        ("LOC", of_loc loc);
+        ("POS", of_pos pos);
+        ("TEST", of_test test);
+        ("EXP", of_exp exp);
+        ("NAME", of_option of_str str)
+      ]
+
+let of_is_rec x =
+  key (match x with IsRec -> "ISREC" | IsNotRec -> "ISNOTREC" )
+
+
+let rec of_ins x =
+  of_list (match x with
+           | Let (loc, bindings)       -> [ key "I_LET"; of_loc loc; of_list_map of_binding bindings ]
+           | Rec (loc, bindings, test) -> [ key "I_REC"; of_loc loc; of_list_map of_binding bindings; of_option of_app_test test ]
+           | InsMatch (loc, exp, clauses, insts)
+             -> [ key "I_INSMATCH"; of_loc loc; of_exp exp; of_list_map of_insclause clauses; of_option (of_list_map of_ins) insts ]
+           | Test ( test, testtype )   -> [ key "I_TEST"; of_app_test test; of_test_type testtype ]
+           | UnShow (loc, strs)        -> [ key "I_UNSHOW"; of_loc loc; of_list_map of_str strs ]
+           | Show (loc, strs)          -> [ key "I_SHOW"; of_loc loc; of_list_map of_str strs ]
+           | ShowAs (loc, exp, str)    -> [ key "I_SHOWAS"; of_loc loc; of_exp exp; of_str str ]
+           | Include (loc, str)        ->
+              if false then
+                [ key "I_INCLUDE"; of_loc loc; of_str str ]
+              else raise (Misc.Fatal "Unexpected include after expansion")
```

Here you can just raise the exception unconditionally, can't you?